rpc/client/client_test.go

```diff
@@ -757,3 +757,62 @@ func TestMempool2Nodes(t *testing.T) {
 
 	assert.Equal(node2.Mempool.TxsBytes(), int64(len("good")))
 }
+
+func TestAggregationLoop(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	mockApp := &mocks.Application{}
+	mockApp.On("InitChain", mock.Anything).Return(abci.ResponseInitChain{})
+	key, _, _ := crypto.GenerateEd25519Key(crand.Reader)
+	node, err := node.NewNode(context.Background(), config.NodeConfig{
+		DALayer:    "mock",
+		Aggregator: true,
+		BlockManagerConfig: config.BlockManagerConfig{
+			BlockTime: 200 * time.Millisecond,
+		}},
+		key, proxy.NewLocalClientCreator(mockApp),
+		&tmtypes.GenesisDoc{
+			ChainID:       "test",
+			InitialHeight: 100,
+			GenesisTime:   time.Now().Local().Add(time.Second * time.Duration(100)),
```

Comment:
The delay value (100 seconds) is passed to `block/manager.go`, but I don't know why this keeps running without waiting. Do you know why it is not waiting?

Reply:
Only the `AggregationLoop` goroutine is sleeping. Other goroutines (like the main goroutine of this unit test) are working normally. Also, using 100s in tests is way too much (IMHO 2s is fine).